recipes/macos.rb

```diff
@@ -0,0 +1,114 @@
+# Install xquartz
+remote_file "/tmp/xquartz.pkg" do
+  source "https://github.com/XQuartz/XQuartz/releases/download/XQuartz-2.8.5/XQuartz-2.8.5.pkg"
+  not_if "pkgutil --pkg-info org.xquartz.X11"
+end
+
+execute "install xquartz" do
+  command "installer -pkg /tmp/xquartz.pkg -target /"
+  not_if "pkgutil --pkg-info org.xquartz.X11"
+end
+
+directory "/Users/jenkins/Library/LaunchAgents" do
+  owner "jenkins"
+  group "staff"
+  recursive true
+end
+
+launchd "org.xquartz.X11.plist" do
+  path "/Users/jenkins/Library/LaunchAgents/org.xquartz.X11.plist"
+  keep_alive true
+  run_at_load true
+  working_directory "/Users/jenkins"
+  process_type "Interactive"
+  program "/Applications/Utilities/XQuartz.app/Contents/MacOS/X11"
+  action [:create, :enable]
+end
+
+
+# Install java
+remote_file "/tmp/jdk8.pkg" do
+  source "https://github.com/adoptium/temurin8-binaries/releases/download/jdk8u362-b09/OpenJDK8U-jdk_x64_mac_hotspot_8u362b09.pkg"
+  not_if "pkgutil --pkg-info net.temurin.8.jdk"
+end
+
+execute "install java" do
+  command "installer -pkg /tmp/jdk8.pkg -target /"
+  not_if "pkgutil --pkg-info net.temurin.8.jdk"
+end
```

Comment:
I've opted not to prematurely add logic to support JDK8 vs JDK11, which is an upcoming transition required on the build farm. I have chosen to keep this PR straightforward and add the JDK11 support in a follow-up.